File: sdks/python/bolt_prediction_sdk/models/knowledge_case_create_request.py
"""
KnowledgeCaseCreateRequest 模型
"""

from typing import Optional, Any, Dict, List
from datetime import datetime
from pydantic import Field

from .base import SDKBaseModel


class KnowledgeCaseCreateRequest(SDKBaseModel):
    """创建案例请求"""

    case_title: str = Field(description="案例标题")
    node_type: Optional[Any] = Field(description="节点类型 bolt/flange", default=None)
    node_id: Optional[Any] = Field(description="节点ID", default=None)
    fault_type: Optional[Any] = Field(description="故障类型", default=None)
    fault_level: Optional[Any] = Field(description="故障级别 1-4", default=None)
    working_condition: Optional[Any] = Field(description="工况信息", default=None)
    sensor_data: Optional[Any] = Field(description="传感器时序数据 [[时间, 数值], ...]", default=None)
    sensor_features: Optional[Any] = Field(description="传感器特征 (58维特征名值对)", default=None)
    diagnosis: Optional[Any] = Field(description="诊断结论", default=None)
    root_cause: Optional[Any] = Field(description="根本原因分析", default=None)
    treatment_plan: Optional[Any] = Field(description="处置方案", default=None)
    effect_evaluation: Optional[Any] = Field(description="效果评估", default=None)
    source_alert_id: Optional[Any] = Field(description="来源告警ID", default=None)
    source_prediction_id: Optional[Any] = Field(description="来源预测记录ID", default=None)
    tags: Optional[Any] = Field(description="标签列表", default=None)
    creator_id: Optional[Any] = Field(description="创建人ID", default=None)
    creator_name: Optional[Any] = Field(description="创建人姓名", default=None)
    tenant_id: Optional[Any] = Field(description="租户ID", default=None)
    submit_for_review: Optional[bool] = Field(description="是否提交审核", default=False)

File: sdks/java/src/main/java/com/boltprediction/sdk/model/KnowledgeCaseCreateRequest.java
package com.boltprediction.sdk.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;
import java.util.Map;
import java.time.OffsetDateTime;

/** 创建案例请求 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class KnowledgeCaseCreateRequest {

    @JsonProperty("case_title")
    private String caseTitle;

    @JsonProperty("node_type")
    private Object nodeType;

    @JsonProperty("node_id")
    private Object nodeId;

    @JsonProperty("fault_type")
    private Object faultType;

    @JsonProperty("fault_level")
    private Object faultLevel;

    @JsonProperty("working_condition")
    private Object workingCondition;

    @JsonProperty("sensor_data")
    private Object sensorData;

    @JsonProperty("sensor_features")
    private Object sensorFeatures;

    @JsonProperty("diagnosis")
    private Object diagnosis;

    @JsonProperty("root_cause")
    private Object rootCause;

    @JsonProperty("treatment_plan")
    private Object treatmentPlan;

    @JsonProperty("effect_evaluation")
    private Object effectEvaluation;

    @JsonProperty("source_alert_id")
    private Object sourceAlertId;

    @JsonProperty("source_prediction_id")
    private Object sourcePredictionId;

    @JsonProperty("tags")
    private Object tags;

    @JsonProperty("creator_id")
    private Object creatorId;

    @JsonProperty("creator_name")
    private Object creatorName;

    @JsonProperty("tenant_id")
    private Object tenantId;

    @JsonProperty("submit_for_review")
    private Boolean submitForReview;

    public String getCaseTitle() {
        return caseTitle;
    }

    public void setCaseTitle(String caseTitle) {
        this.caseTitle = caseTitle;
    }

    public Object getNodeType() {
        return nodeType;
    }

    public void setNodeType(Object nodeType) {
        this.nodeType = nodeType;
    }

    public Object getNodeId() {
        return nodeId;
    }

    public void setNodeId(Object nodeId) {
        this.nodeId = nodeId;
    }

    public Object getFaultType() {
        return faultType;
    }

    public void setFaultType(Object faultType) {
        this.faultType = faultType;
    }

    public Object getFaultLevel() {
        return faultLevel;
    }

    public void setFaultLevel(Object faultLevel) {
        this.faultLevel = faultLevel;
    }

    public Object getWorkingCondition() {
        return workingCondition;
    }

    public void setWorkingCondition(Object workingCondition) {
        this.workingCondition = workingCondition;
    }

    public Object getSensorData() {
        return sensorData;
    }

    public void setSensorData(Object sensorData) {
        this.sensorData = sensorData;
    }

    public Object getSensorFeatures() {
        return sensorFeatures;
    }

    public void setSensorFeatures(Object sensorFeatures) {
        this.sensorFeatures = sensorFeatures;
    }

    public Object getDiagnosis() {
        return diagnosis;
    }

    public void setDiagnosis(Object diagnosis) {
        this.diagnosis = diagnosis;
    }

    public Object getRootCause() {
        return rootCause;
    }

    public void setRootCause(Object rootCause) {
        this.rootCause = rootCause;
    }

    public Object getTreatmentPlan() {
        return treatmentPlan;
    }

    public void setTreatmentPlan(Object treatmentPlan) {
        this.treatmentPlan = treatmentPlan;
    }

    public Object getEffectEvaluation() {
        return effectEvaluation;
    }

    public void setEffectEvaluation(Object effectEvaluation) {
        this.effectEvaluation = effectEvaluation;
    }

    public Object getSourceAlertId() {
        return sourceAlertId;
    }

    public void setSourceAlertId(Object sourceAlertId) {
        this.sourceAlertId = sourceAlertId;
    }

    public Object getSourcePredictionId() {
        return sourcePredictionId;
    }

    public void setSourcePredictionId(Object sourcePredictionId) {
        this.sourcePredictionId = sourcePredictionId;
    }

    public Object getTags() {
        return tags;
    }

    public void setTags(Object tags) {
        this.tags = tags;
    }

    public Object getCreatorId() {
        return creatorId;
    }

    public void setCreatorId(Object creatorId) {
        this.creatorId = creatorId;
    }

    public Object getCreatorName() {
        return creatorName;
    }

    public void setCreatorName(Object creatorName) {
        this.creatorName = creatorName;
    }

    public Object getTenantId() {
        return tenantId;
    }

    public void setTenantId(Object tenantId) {
        this.tenantId = tenantId;
    }

    public Boolean getSubmitForReview() {
        return submitForReview;
    }

    public void setSubmitForReview(Boolean submitForReview) {
        this.submitForReview = submitForReview;
    }

}